oauthproxy.go

```diff
@@ -429,9 +429,16 @@ func (p *OAuthProxy) SetCSRFCookie(rw http.ResponseWriter, req *http.Request, va
 // ClearSessionCookie creates a cookie to unset the user's authentication cookie
 // stored in the user's session
 func (p *OAuthProxy) ClearSessionCookie(rw http.ResponseWriter, req *http.Request) {
-	cookies := p.MakeSessionCookie(req, "", time.Hour*-1, time.Now())
-	for _, clr := range cookies {
-		http.SetCookie(rw, clr)
+	reqCookies := req.Cookies()
+	var cookies []*http.Cookie
+
+	for _, c := range reqCookies {
+		if strings.HasPrefix(c.Name, p.CookieName) {
```

Comment:
Is checking for a prefix sufficient? Perhaps we could use a regex to check if the cookie is of the form `<name>_<index>` to be safer?

Reply:
I switched to regex with commit f1db10e